### DISCUSSION TITLE
create a settings file that can be imported into new organs

### DISCUSSION THREAD

**Discussion by davidgritter** [GrandOrgue/grandorgue#1951] [Help]
I run Grandorgue headless, and so have recently intalled a small Nextion display on my console(see pictures.)

The nextion display communicates to GrandOrgue over midi, and many of its functions interact with elements on the built in MASTER CONTROLS and GENERALS panel.  These interactions are the same for all newly configured sample sets.  

I am trying to figure out a way to export the configuration of these elements to a settings file so that they can be imported into a different organ without modifying other pre-existing settings in that organ, or at least be imported before other settings are configured.  I had thought the the .cmb files used to be human readable and editable, but this appears to no longer be the case, so that option is not on the table.  Any suggestions?
![SSCN2902](https://github.com/user-attachments/assets/565386af-d468-479f-8942-19d32f59dbb9)
![SSCN2903](https://github.com/user-attachments/assets/477eeeff-70cb-4265-b50c-6e20c2058653)
![SSCN2904](https://github.com/user-attachments/assets/344d9d91-e5ab-442a-b203-cb6f28c47457)

**Comment by thbeckpr** [comment]
I have a similar situation with my setup. I written some apps to respond to hardware and send MIDI messages to the Combination Setter panel as well as others. 

Every time a create/use a new sample sets I have to tediously set each of the elements on the panel to respond to the appropriate MIDI message. 

It would be nice to be able to set these as defaults at the application level rather than at the organ level. 

Is this possible?

**Comment by larspalo** [comment]
Some of the elements already can have their connections/messages pre-defined on the Initial MIDI tab of File->Settings. If set, they will be configured for every organ at first opening. If more existing elements are desired that are generally available to all organs then I suggest you create an Issue as a feature request for them to be added as possible to configure.

**Reply by davidgritter** [reply]
I would consider doing this if the export settings approach won't work.   Here is what I suggest:

Include the significant buttons from the generals and divisional panels

Include the SET button as well as the General Cancel

Include the SAVE button.

Include the ability to define midi send messages as well as receive -currently it doesn't appear that any of the existing elements allow definition of midi sends.

Include all of the active labels so that they can send hauptwerk sysex messages.

Does anyone have anything else that should be added?  Perhaps some of the couplers, although these might be organ specific.

**Comment by davidgritter** [comment]
I have experimented with this a bit by clearing the settings in an organ, then manually inputting the settings for the standard panels that I want to export.  I then export the settings.  but when I load these setting into another organ, it tries to import all sorts of things related to rankxxx,pipexxx like timings, release loadings release tail.  These are all things that  I have never entered into any organ, so clearly the export settings function is exporting settings that I as a user have never entered as settings. Why?  Are they things that are in the ODF that really shouldn't be exported as settings or is something else going on?

So then I created an odf with just a single panel, keyboard, and windchest,  but no stops, ranks, pipes, expresssions or switches.  I then re-enter manually the settings for the standard panels and export settings.  Now importing the settings file no longer results in warning messages about odd pseudo-settings, so this may  be a workable solution.  However I did also note that if I assign midi devices and channels for keyboards to listen on, importing this setting file resets these settings completely so that they need to be re-entered.

**Reply by larspalo** [reply]
@davidgritter The settings files .cmb are unique to each organ and should **not** be imported into other organs as they contain "everything". That's why we're working on (planning) other means to do configurations that are not so strictly tied to specific organs. Initial MIDI would be a great tool if it was expanded appropriately.

As always, there are ways to manually transfer settings from one .cmb to another with a lot of manual labor and ingenuity (like you're hinting at), but that's definitely not supported, nor a desired way to do things.
